Source/WebCore/html/canvas/WebGLRenderingContextBase.h

```diff
@@ -1067,6 +1062,41 @@ class WebGLRenderingContextBase : public GraphicsContextGL::Client, public GPUBa
     WeakPtrFactory<WebGLRenderingContextBase> m_contextObjectWeakPtrFactory;
 };
 
+template<typename T>
+bool WebGLRenderingContextBase::validateWebGLObject(const char* functionName, const T& object)
```

Comment:
Should SFINAE be used to only enable these functions for types `T` that are derived from `WebGLObject`?

Reply:
Well, the idea is to be able to remove the dependency to the base class..
I think the other idea is that if you *want* to validate with non-WebGLObject, that working could be seen as a feature.
That being said, passing wrong object tends to give bad error messages so from that perspective some typing would be good..